ioctl/cmd/account/accountcreate2.go

```diff
@@ -0,0 +1,239 @@
+		config.Chinese: "为ioctl创建新账户通过助记词",
+	}
+	createByMnemonicCmdUses = map[config.Language]string{
+		config.English: "create2",
+		config.Chinese: "create2 创建",
+	}
+
+	mnemonicConfigFilePath = filepath.Join(config.ReadConfig.Wallet, "mnemonic.yml")
+)
+
+// accountCreateCmd represents the account create command
+var accountCreateByMnemonicCmd = &cobra.Command{
+	Use:   config.TranslateInLang(createByMnemonicCmdUses, config.UILanguage),
+	Short: config.TranslateInLang(createByMnemonicCmdShorts, config.UILanguage),
+	Args:  cobra.ExactArgs(0),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		err := accountCreateByMnemonic()
+		return output.PrintError(err)
+	},
+}
+
+type mnemonicConfig struct {
+	PrivateKey []byte `yaml:"k"`
+	Mnemonic   []byte `yaml:"m"`
+	Password   []byte `yaml:"p"`
+	ID         uint32 `yaml:"i"`
+}
+
+func (c *mnemonicConfig) DecryptMnemonic() ([]byte, error) {
+
+	privateKey, err := x509.ParsePKCS1PrivateKey(c.PrivateKey)
+	if err != nil {
+		return nil, err
+	}
+	return privateKey.Decrypt(nil, c.Mnemonic, &rsa.OAEPOptions{Hash: crypto.SHA256})
+}
+
+func (c *mnemonicConfig) DecryptPassword() ([]byte, error) {
+
+	privateKey, err := x509.ParsePKCS1PrivateKey(c.PrivateKey)
+	if err != nil {
+		return nil, err
+	}
+	return privateKey.Decrypt(nil, c.Password, &rsa.OAEPOptions{Hash: crypto.SHA256})
+}
+
+func hashECDSAPublicKey(publicKey *ecdsa.PublicKey) []byte {
+	k := ecrypto.FromECDSAPub(publicKey)
+	h := hash.Hash160b(k[1:])
+	return h[:]
+}
+
+func init() {
+	accountCreateByMnemonicCmd.Flags().UintVarP(&numAccounts, "num", "n", 1,
+		config.TranslateInLang(flagNumUsages, config.UILanguage))
+}
+
+func createUserByMnemonic() error {
+	output.PrintQuery("Set password\n")
+	password, err := util.ReadSecretFromStdin()
+	if err != nil {
+		return output.NewError(output.InputError, "failed to get password", err)
+	}
+	output.PrintQuery("Enter password again\n")
+	passwordAgain, err := util.ReadSecretFromStdin()
+	if err != nil {
+		return output.NewError(output.InputError, "failed to get password", err)
+	}
+	if password != passwordAgain {
+		return output.NewError(output.ValidationError, ErrPasswdNotMatch.Error(), nil)
+	}
+
+	entropy, _ := bip39.NewEntropy(128)
+	mnemonic, _ := bip39.NewMnemonic(entropy)
+
+	output.PrintResult(fmt.Sprintf("Mnemonic pharse: %s\n"+
+		"Save them somewhere safe and secret.", mnemonic))
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return err
+	}
+
+	publicKey := privateKey.PublicKey
+
+	encryptedMnemonic, err := rsa.EncryptOAEP(
+		sha256.New(),
+		rand.Reader,
+		&publicKey,
+		[]byte(mnemonic),
+		nil)
+	if err != nil {
+		return err
+	}
+
+	encryptedPassword, err := rsa.EncryptOAEP(
+		sha256.New(),
+		rand.Reader,
+		&publicKey,
+		[]byte(password),
+		nil)
+	if err != nil {
+		return err
+	}
+
+	out, err := yaml.Marshal(&mnemonicConfig{
+		PrivateKey: x509.MarshalPKCS1PrivateKey(privateKey),
```

Comment:
think we need to encrypt `privateKey` using `password` as key before marshal to PKCS1?
otherwise if the config file is copied/leaked, mnemonic can be recovered

Reply:
also, i think we don't need to create an extra RSA-2048 key?
can just use `password` to encrypt the mnemonic words and save that, that's it